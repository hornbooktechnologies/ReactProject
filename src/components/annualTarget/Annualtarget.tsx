import React,{ useState, useEffect } from "react";
import { useTranslation } from "react-i18next";
import { DpmService } from "../../services/DpmServices";
import { AppConstant } from "../../utils/AppConstants";
import LoadingError from "../common/LoadingError";
import Spinner from "../common/Spinner";
import PageHeader from "../common/PageHeader";
import GrossMarginTab from "./annualTargetTab/GrossMarginTab";
import OverviewTab from "./annualTargetTab/OverviewTab";
import OpexTab from "./annualTargetTab/OpexTab";
import CustomTabs from "../common/CustomTabs";
import { CustomTabModel } from "../../models/CustomTabModel";
import { AnnualTargetDataModel } from "../../models/annualTarget/AnnualTargetDataModel";
import { ApiParamUtils } from '../../utils/apiUtils/ApiParamUtils'
import "../../assets/scss/annualTarget/_annualstatus.scss";

const AnnualTarget = () => {
  const { i18n } = useTranslation();
  const [annualTargetData, setAnnualTargetData] = useState({});
  const [AnnualTargetSelectedTab, AnnualTargetClickedTab] = useState(
    AppConstant.OVERVIEW
  );
  const [epochSeconds] = useState(Math.trunc(new Date().getTime() / 1000));
  const [errorThrown, seterrorThrown] = useState(null);
 
  useEffect(() => {
    DpmService.fetchAnnualTarget(ApiParamUtils.plantName(), ApiParamUtils.unitName(), epochSeconds)
      .then((res) => {
        setAnnualTargetData(res.data);
      })
      .catch((error) => {
        seterrorThrown(error);
        setAnnualTargetData("");
      });
  }, [epochSeconds, errorThrown]);

  const onTargetClick = (data: string) => {
    AnnualTargetClickedTab(data);
  };
  const tabData: CustomTabModel = {
    selectedTab: AnnualTargetSelectedTab,
    onTabSelect: onTargetClick,
    tabs: [
      {
        tabId: "Overview",
        i8nCode: "ANNUAL_TARGET.OVERVIEW",
      },
      {
        tabId: "GrossMargin",
        i8nCode: "ANNUAL_TARGET.GROSS_MARGIN_BUTTON",
      },
      {
        tabId: "Opex",
        i8nCode: "LABELS.OPEX",
      },
    ],
  };

  const tabPropsData: AnnualTargetDataModel = {
    AnnualTargetSelectedTab: AnnualTargetSelectedTab,
    jsonData: annualTargetData,
  };

  if (Object.entries(annualTargetData).length > 0) {
    return (
      <>
        <PageHeader
          title={i18n.t("ANNUAL_TARGET.TITLE")}
          description={i18n.t("ANNUAL_TARGET.DESCRIPTION")}
        />
        <CustomTabs {...tabData} />
        <ul className="uk-switcher mt-24">
          {(() => {
            switch (AnnualTargetSelectedTab) {
              case AppConstant.OVERVIEW:
                return <OverviewTab {...tabPropsData} />;
              case AppConstant.GROSSMARGIN:
                return <GrossMarginTab {...tabPropsData} />;
              case AppConstant.OPEX:
                return <OpexTab {...tabPropsData} />;
              default:
                return <OverviewTab {...tabPropsData} />;
            }
          })()}
        </ul>
      </>
    );
  } else if (errorThrown != null) {
    return <LoadingError />;
  } else {
    return <Spinner />;
  }
};
export default AnnualTarget;
