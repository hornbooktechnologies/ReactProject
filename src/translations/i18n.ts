import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
