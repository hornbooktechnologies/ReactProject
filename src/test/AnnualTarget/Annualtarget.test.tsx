import React from "react";
import renderer from "react-test-renderer";
import Annualtarget from "../../components/annualTarget/Annualtarget";
import { I18nextProvider } from "react-i18next";
import i18n from "../../i18nForTests";
describe("Component: Annualtarget", () => {
  it("Annualtarget renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <Annualtarget />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
