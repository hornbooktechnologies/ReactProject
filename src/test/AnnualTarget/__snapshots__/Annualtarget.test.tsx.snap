// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Annualtarget Annualtarget renders correctly 1`] = `
<div
  className="spinner"
>
  <div
    className="color-white"
  >
    <div
      uk-spinner="ratio: 3"
    />
  </div>
</div>
`;
